### ISSUE TITLE
Setting progress on restored state LottieAnimationView with async composition does not work

### ISSUE THREAD

**Issue by elihart** [airbnb/lottie-android#975]
**The issue:** A LottieAnimationView (in a recyclerview) does not show the correct frame when restoring state, when the animation is not playing and the composition is loaded async.

In my case the animation has already played and I want progress to be reset to 1 to show the final frame of the animation. I don't want it to play again, but instead freeze on that last frame. Instead, nothing is drawn. I am loading the animation from a url.
 
This one is a bit tricky, but I think I've narrowed down what is going on. Lottie correctly queues the callback to set progress to 1, loads the composition, runs the callback to set progress to 1, and then sets the frame to the last frame. However, `LottieValueAnimator#setFrame` never calls `notifyUpdate` and the  drawable/composition never draws the frame.

This is because the animator's `frame` field was on the same frame when state was saved, and restoring it to the same value is a no-op
```
public void setFrame(int frame) {
    if (this.frame == frame) {
      return;
    }
```

So nothing ever gets drawn.

This doesn't happen if the composition is loaded from a local file, i'm guessing because it hits the synchronous memory cache for loading the composition and something is slightly different.

**Comment by gpeal** [comment]
@elihart Can you try this again on master? There have been a number of fixes to various things.

**Comment by elihart** [comment]
I kind of forget where I was having this problem, but if the url loading hits the memory cache in the same way now it is probably fixed 👍

**Comment by gpeal** [comment]
Nice! Will close this for now. Let me know if this comes back up. I published 3.0.0-beta2. It should be as stable as any other release but has a *ton* of fixes so try it out!
